File: src/main/java/top/wjc/bookmallbackend/entity/Book.java
package top.wjc.bookmallbackend.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
/**
 * 图书实体。
 *
 * <p>对应数据库中的 `book` 表，保存图书基础信息、库存、价格以及乐观锁版本号。
 */
public class Book {
    private Long id;
    private Integer version;
    private String bookName;
    private String author;
    private String publisher;
    private String isbn;
    private Long categoryId;
    private BigDecimal price;
    private Integer stock;
    private String coverImage;
    private String description;
    private Integer status;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: frontend/src/api/admin/book.ts
import request from '../request'
import type { AdminBookListItem, PageResult } from '../../types/api'

export type AdminBookListParams = {
  page?: number
  pageSize?: number
  categoryId?: number
  keyword?: string
}

export type BookCreateRequest = {
  bookName: string
  author?: string
  publisher?: string
  isbn?: string
  categoryId: number
  price: number
  stock: number
  coverImage?: string
  description?: string
  status: number
}

export type BookUpdateRequest = BookCreateRequest

export type BookStatusRequest = {
  status: number
}

export const adminBookApi = {
  list: (params?: AdminBookListParams) => request.get<PageResult<AdminBookListItem>>('/api/admin/books', { params }),
  create: (payload: BookCreateRequest) => request.post<void>('/api/admin/books', payload),
  update: (id: number | string, payload: BookUpdateRequest) => request.put<void>(`/api/admin/books/${id}`, payload),
  remove: (id: number | string) => request.delete<void>(`/api/admin/books/${id}`),
  updateStatus: (id: number | string, payload: BookStatusRequest) =>
    request.put<void>(`/api/admin/books/${id}/status`, payload),
  upload: (file: File) => {
    const formData = new FormData()
    formData.append('file', file)
    return request.post<string>('/api/admin/upload/book-cover', formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
  },
}

export default adminBookApi
